kafka/src/kafka/consumer_component.cpp

```diff
@@ -71,6 +71,10 @@ additionalProperties: false
             consumer group id.
             Topic partition evenly distributed
             between consumers with the same `group_id`
+    client_id:
+        type: string
+        description: An id string to pass to the server when making requests
```

Comment:
This description is not very informative, imho.
Better to copy description from https://github.com/confluentinc/librdkafka/blob/master/CONFIGURATION.md and describe some usecases (as you texted in the PR description)

Reply:
And say that client.id may me an arbitrary string

Reply:
Change